software/firmware/calibrate.py

```diff
@@ -44,35 +71,36 @@ def wait_for_voltage(voltage):
 else:
     for voltage in range(11):
         readings.append(wait_for_voltage(voltage))
-
+        
 with open(f'lib/calibration.py', 'w') as file:
     values = ", ".join(map(str, readings))
     file.write(f"INPUT_CALIBRATION_VALUES=[{values}]")
-print(f'\n{readings}\n')
+    
 
+from europi import AnalogueInput
+ain = AnalogueInput(26)
+cv1 = PWM(Pin(21))
 
-## Output Calibration
+oled.centre_text(f'Plug CV1 into\nanalogue in\nthen press B1')
+wait_for_b1(1)
+oled.centre_text('Calibrating...')
 
 output_duties = [0]
 duty = 0
 reading = 0
-step = 25
-input('\nPlease plug CV output 1 into the analogue input and then press Enter')
 for voltage in range(1,11):
     while abs(reading - voltage) > 0.005 and reading < voltage:
-        if duty > MAX_UINT16:
```

Comment:
It looks like the changes from #46 got overwritten by this change. Most important is the check here to prevent users from getting stuck in an infinite loop if their module is unable to reach 10v.